tests/assembly/riscv64gc-zbb.rs

```diff
@@ -0,0 +1,23 @@
+//@ assembly-output: emit-asm
+//@ compile-flags: --target riscv64gc-unknown-linux-gnu
+//@ compile-flags: -C target-feature=+zbb
+//@ compile-flags: -C opt-level=3
+//@ needs-llvm-components: riscv
+
+#![crate_type = "lib"]
+
+#[no_mangle]
+// CHECK-LABEL: issue_114508_u32:
+pub fn issue_114508_u32(a: u32, b: u32) -> u32 {
+    // CHECK:      maxu a0, a0, a1
+    // CHECK-NEXT: ret
+    u32::max(a, b)
+}
+
+#[no_mangle]
+// CHECK-LABEL: issue_114508_i32:
+pub fn issue_114508_i32(a: i32, b: i32) -> i32 {
+    // CHECK:      max a0, a0, a1
+    // CHECK-NEXT: ret
+    i32::max(a, b)
+}
```

Comment:
as mentioned in the issue, this problem affects both loongarch and riscv, and it affects riscv both with and without zbb, having consulted Godbolt: https://godbolt.org/z/hnWhx5bz9

so I'd like to see two more tests. at least one should be another revision in this test, specifically for riscv-without-zbb (well, this revision should become riscv-zbb, actually). you can make the loongarch test its own file or yet another revision.

Reply:
I have added non-zbb ext checks in it and improved conditions (with all `-NEXT` tag).

Since RISC-V and LoongArch are different archs (though highly similar), checking the expected LoongArch ASM output requires we do the similar adjustment to Rust ABI. I have added it in the third commit.